front/lib/resources/content_fragment_resource.ts

```diff
@@ -276,3 +284,102 @@ export async function getContentFragmentText({
 
   return getPrivateUploadBucket().fetchFileContent(filePath);
 }
+
+async function getSignedUrlForRawContentFragment({
+  workspaceId,
+  conversationId,
+  messageId,
+}: {
+  workspaceId: string;
+  conversationId: string;
+  messageId: string;
+}): Promise<string> {
+  const fileLocation = fileAttachmentLocation({
+    workspaceId,
+    conversationId,
+    messageId,
+    contentFormat: "raw",
+  });
+
+  return getPrivateUploadBucket().getSignedUrl(fileLocation.filePath);
+}
+
+export async function renderContentFragmentForModel(
+  message: ContentFragmentType,
+  conversation: ConversationType,
+  model: ModelConfigurationType,
+  {
+    excludeImages,
+  }: {
+    excludeImages: boolean;
+  }
+): Promise<Result<ContentFragmentMessageTypeModel, Error>> {
+  const { contentType, sId, title } = message;
+
+  try {
+    if (isSupportedImageContentFragmentType(contentType) && !excludeImages) {
+      if (!model.supportsVision) {
+        return new Ok({
+          role: "content_fragment",
+          name: `inject_${contentType}`,
+          content: [
+            {
+              type: "text",
+              text: `<attachment type="${contentType} title="${title}">[Image content interpreted by a vision-enabled model. Detailed description not available in this context.]`,
```

Comment:
@spolu if you wanna double check the placeholder for model without vision support 🙏.

Reply:
Looks reasonable 👍

I would remove the "Detailed" which serves no purpose to save a token or 2 :p